phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -612,8 +612,8 @@ impl TestVm {
     async fn send_serial_str_async(&self, string: &str) -> Result<()> {
         let mut bytes = Vec::new();
         bytes.extend_from_slice(string.as_bytes());
-        self.send_serial_bytes_async(bytes).await?;
-        self.send_serial_bytes_async(vec![b'\n']).await
+        bytes.extend_from_slice(&[b'\n']);
```

Comment:
nitpicky, not a big deal: since this is a single character, why do we need to call `extend_from_slice` with a single-character-long slice? can't we just write:
```suggestion
        bytes.push(b'\n');
```